lib/salus/scanners/yarn_audit.rb

```diff
@@ -7,14 +7,84 @@
 
 module Salus::Scanners
   class YarnAudit < NodeAudit
-    AUDIT_COMMAND = 'yarn audit --json'.freeze
+    # the command was previously 'yarn audit --json', which had memory allocation issues
+    # see https://github.com/yarnpkg/yarn/issues/7404
+    AUDIT_COMMAND = 'yarn audit --no-color'.freeze
+    TT = "┌───────────────┬──────────────────────────────────────────────────────────────┐".freeze
+    TB = "└───────────────┴──────────────────────────────────────────────────────────────┘".freeze
 
     def should_run?
       @repository.yarn_lock_present?
     end
 
+    def run
+      shell_return = Dir.chdir(@repository.path_to_repo) do
+        command = "#{AUDIT_COMMAND} #{scan_deps}"
+        shell_return = run_shell(command)
+
+        excpts = @config.fetch('exceptions', []).map { |e| e["advisory_id"].to_i }
+        report_info(:ignored_cves, excpts)
+        return report_success if shell_return.success?
+
+        stdout_lines = shell_return.stdout.split("\n")
+        table_start_pos = stdout_lines.index(TT)
+        table_end_pos = stdout_lines.rindex(TB)
+        table_lines = stdout_lines[table_start_pos..table_end_pos]
+        # lines contain 1 or more vuln tables
+
+        # if no table in output
+        if table_start_pos.nil? || table_end_pos.nil?
+          report_error(shell_return.stderr, status: shell_return.status)
+          report_stderr(shell_return.stderr)
+          return report_failure
+        end
+
+        vulns = parse_output(table_lines)
+        vuln_ids = vulns.map { |v| v['id'] }
+        report_info(:vulnerabilities, vuln_ids.uniq)
+
+        vulns.reject! { |v| excpts.include?(v['id']) }
+        # vulns were all whitelisted
+        return report_success if vulns.empty?
+
+        log(format_vulns(vulns))
+        report_stdout(vulns.to_json)
+        report_failure
+      end
+    end
+
     private
 
+    def parse_output(lines)
+      empty_cell = "│               │"
+      vulns = []
+
+      i = 0
+      while i < lines.size
+        if lines[i] == TT
+          vuln = {}
+        elsif lines[i] =~ /^\│ [A-Za-z]/
+          # line has attr name and val, like
+          # | Path          | eslint > file-entry-cache > flat-cache > write > mkdirp >    |
+          line_split = lines[i].split("│")
+          key = line_split[1].strip
+          val = line_split[2].strip
+          vuln[key] = val
+        elsif lines[i].start_with?(empty_cell)
```

Comment:
Instead of looking for lines with `empty_cell`, perhaps you could just check if the `key` is empty after you do `.strip` on it?

Reply:
Can we also add a test to make sure we are capturing multi-line values correctly?

Reply:
Good suggestion. I got rid of the `empty_cell`.

Reply:
The spec test on `failure-2` dir already covers multi-line values.
If you run `yarn audit` directly on the dir, it shows a multi-line value.